index.js

```diff
@@ -84,40 +114,103 @@ module.exports = {
         'eslint-plugin-ember@latest',
         'eslint-plugin-n@latest',
         '@babel/eslint-parser@latest',
+        'ember-template-lint@latest',
+
+        '@ember/string@latest',
         'ember-resolver@latest',
         'ember-load-initializers@latest',
+        'qunit@latest',
+        'qunit-dom@latest',
+        'stylelint@latest',
+        'stylelint-prettier@latest',
+        'stylelint-config-standard@latest',
+        'concurrently@latest',
+
         // Needed for eslint
         'globals',
         'babel-plugin-ember-template-compilation',
         'prettier-plugin-ember-template-tag',
+
+        // Dependencies out of date from upstream
+        // (or waiting for the release train to catch up)
+        'ember-route-template',
+
+        // TypeScript
+        // Note that Vite supports TypeScript with 0 configuration on the user's part
+        ...(options.typescript
+          ? [
+              '@babel/plugin-transform-typescript',
+              // TODO: see if there is a better way we can
+              //       sync these libraries' versions
+              // typescript-eslint doesn't support typescript 5.6 yet
+              'typescript@~5.5.0',
+              '@glint/core@unstable',
+              '@glint/environment-ember-loose@unstable', // currently required :(
+              '@glint/environment-ember-template-imports@unstable',
+              '@glint/template@unstable',
+              '@types/eslint__js',
+              'typescript-eslint@latest',
+              '@typescript-eslint/eslint-plugin@latest',
+              '@typescript-eslint/parser@latest',
+            ]
+          : []),
       ],
       packageManager: options.packageManager,
     });
   },
 
   async afterInstall(options) {
     const filesToDelete = [
+      // now in the project root
       'app/index.html',
-      // replaced with the new ESLint flat config
+      // replaced with application.gjs/gts
```

Comment:
let's not do this until a follow up PR

Reply:
about to get simpler: https://github.com/emberjs/rfcs/pull/1046

Reply:
resolving, because I don't want to set up the glint template registry in here :see_no_evil: